Sources/StringsLintFramework/Parsers/StructuredPlaceholderCommentParser.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  StructuredPlaceholderCommentParser.swift
+//  
+//
+//  Created by Mark Hall on 2022-07-19.
+//
+
+import Foundation
+
+public struct StructuredPlaceholderCommentParser: LocalizableParser {
+
+    enum StructuredPlaceholderCommentParserError: Error {
+        case missingHeaderSeparatorComment
+    }
+
+    public static var identifier: String {
+        return "structured_placeholder_comment_parser"
+    }
+
+    public var supportedFileExtentions: [String] {
+        return [ "strings" ]
+    }
+
+    public init() {
+    }
+
+    public init(configuration: Any) throws {
+        // Parser does not support any configuration
+        self.init()
+    }
+
+    //TODO: (Mark Hall, July 18) only need to support the english file
+    public func support(file: File) -> Bool {
+        return file.name.hasSuffix(".strings")
+    }
+
+    public func parse(file: File) throws -> [LocalizedString] {
+
+        let tableName = file.name.bridge().deletingPathExtension
+        let locale = Locale(url: file.url)
+
+        var strings = [LocalizedString]()
+        guard file.content.contains("//--START CONTENT--") else {
```

Comment:
what is this?

Reply:
see other comment about this in the strings file